### PR TITLE
[IMP] account_check_printing: make check printing generic

### PR DESCRIPTION
**Pull request by rgo-odoo**
Task: https://www.odoo.com/web#id=2030714&action=327&model=project.task&view_type=form&menu_id=4720
Pad: https://pad.odoo.com/p/r.ddf719c3c5c3d5e68c1523369961d6b3

### CODE REVIEW

File: addons/account_check_printing/models/account_payment.py [hunk 1]
```diff
@@ -96,8 +96,20 @@ def unmark_sent(self):
         self.write({'state': 'posted'})
 
     def do_print_checks(self):
-        """ This method is a hook for l10n_xx_check_printing modules to implement actual check printing capabilities """
-        raise UserError(_("You have to choose a check layout. For this, go in Apps, search for 'Checks layout' and install one."))
+        if not self:
+            raise _('Please select a record to print check.')
```
Comment (jco-odoo) [inline addons/account_check_printing/models/account_payment.py]: How is this different from any other method that we put this raise?  (as other methods could have an empty self then too)

File: addons/account_check_printing/models/res_company.py [hunk 2]
```diff
@@ -5,16 +5,11 @@
...
-        selection=[
-            ('disabled', 'None'),
-            ('action_print_check_top', 'check on top'),
-            ('action_print_check_middle', 'check in middle'),
-            ('action_print_check_bottom', 'check on bottom')
-        ],
-        default="action_print_check_top")
+    # This field needs to be overridden with `selection_add` in the modules which intends to add report layouts.
+    # The xmlID of all the report actions which are actually Check Layouts has to be kept as key of the selection.
+    account_check_printing_report_actions = fields.Selection(selection=[
```
Comment (jco-odoo) [inline addons/account_check_printing/models/res_company.py]: Can we keep the old name?  It is also only one report_action.

File: addons/account_check_printing/models/account_payment.py (hunk 1 above)
Comment (rgo-odoo) [reply addons/account_check_printing/models/account_payment.py]: @jco-odoo I can't recall why I did this but it looks like it is not needed, hence removing it.

**Comment by qdp-odoo** [comment]
@robodoo r+

**Pull request closed without merging**